File: client/node/example/quick/testFinger.js
const finger = require('../finger');
const fs = require('fs');

const TEMPLATE_FORMAT = finger.TEMPLATE_FORMAT_SUPREMA;
const IMAGE_FILENAME = './finger.bmp';

function test(devID) {
	console.log('>>> Scan a finger...');

  return finger.scan(devID, TEMPLATE_FORMAT, 50)
  .then((templateData) => {
    console.log('Template: %d bytes', templateData.length);
    return finger.getImage(devID);
  })
  .then((bmpImage) => {
    fs.writeFileSync(IMAGE_FILENAME, bmpImage);
    return finger.getConfig(devID);
  })
  .then((config) => {
    console.log('Finger config: ', config);
  })
  .catch((err) => {
    console.error('Cannot finish the finger test: ', err);
    throw err;
  });
}

module.exports.test = test;

File: client/python/example/quick/testFinger.py
import grpc
import finger_pb2

from example.finger.finger import FingerSvc

QUALITY_THRESHOLD = 50
IMAGE_FILENAME = './finger.bmp'

def testFinger(fingerSvc, deviceID):
  try:
    print('>>> Scan a finger...', flush=True)

    templateData = fingerSvc.scan(deviceID, finger_pb2.TEMPLATE_FORMAT_SUPREMA, QUALITY_THRESHOLD)
    print(f'Template data: {templateData}', flush=True)

    fingerImage = fingerSvc.getImage(deviceID)
    f = open(IMAGE_FILENAME, 'wb')
    f.write(fingerImage)
    f.close()

    fingerConfig = fingerSvc.getConfig(deviceID)
    print(f'\nFingerprint config: \n{fingerConfig}', flush=True)
 
  except grpc.RpcError as e:
    print(f'Cannot complete the finger test: {e}')
